src/test/ui/privacy/public-instead-of-pub.rs

```diff
@@ -0,0 +1,4 @@
+// Checks what happens when `public` is used instead of the correct, `pub`
```

Comment:
Could you move the test to `src/test/ui/parser`? As much as the parsing change being made here relates to the privacy keywords, I think `src/test/ui/privacy`'s tests are more about the part of the compiler that checks those.